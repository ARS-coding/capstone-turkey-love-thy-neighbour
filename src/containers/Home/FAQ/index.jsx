import React from "react";

// import { Container } from "react-bootstrap";
import "./index.scss";

function FAQ() {
  return (
    <section>
      <p>FAQ</p>
    </section>
  );
}

export default FAQ;
