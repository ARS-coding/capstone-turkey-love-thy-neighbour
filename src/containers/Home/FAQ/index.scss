@import "../../../style/variables";

  h3{
    color:  $logo-blue;
    font-size: max(15px);
    padding-top: 20px;
    padding-bottom: 10px;
    
    font-family: Roboto, sans-serif;
  }

  h2{
    color:  $logo-red;
    font-weight: bold;
    font-family: Roboto, sans-serif;
  }

  p {
    color: black;
    font-size: 15px;
    font-family: Roboto, sans-serif;
  }

  img {
    width: 100%;
    max-height: 450px;
    @media screen and (max-width: 767px) {
      max-height: 280px;
    }
  }

  .first-column {
    @media screen and (max-width: 768px) {
      padding: 30px !important;
      
    }
  }

