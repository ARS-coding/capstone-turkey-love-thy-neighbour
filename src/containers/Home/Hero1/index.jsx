import React from "react";

import { Container, Row, Col } from "react-bootstrap";

import styled from "styled-components";

import "./index.scss";

function Hero1() {
  const CATButton = styled.button`
    background-color: transparent;
    border: none;
    color: white;
    padding: 13px 65px;
    margin-top: 13px;
  `;

  return (
    <section className="hero1-img-bg container-fluid">
      <Container className="hero1-content-container">
        <Row className="hero1-content-container">
          <Col className="pe-0 d-flex flex-column justify-content-center align-items-start">
            <h1 className="fs-1">Love Thy Neighbor!</h1>
            <p>
              Letâ€™s build a strong, dynamic and caring community that uplifts
              each other.
            </p>
            <CATButton className="cta-button">Join Us!</CATButton>
          </Col>
          <Col className="ps-0 right-column">a</Col>
        </Row>
      </Container>
    </section>
  );
}

export default Hero1;
