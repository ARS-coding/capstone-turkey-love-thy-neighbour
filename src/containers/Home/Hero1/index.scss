@import "../../../style/variables";

.hero1-img-bg,
.hero1-content-container,
.hero1-content-row {
  height: calc(100vh - 60px);
}

.hero1-img-bg {
  background-image: url("../../../utils/Hero1BgImg.png");

  h1 {
    color: $logo-blue;
    font-size: calc(5vw - 10px);
    // font-size: 65px;
  }

  p {
    color: white;
    font-size: 32px;
  }

  .cta-button {
    background-color: $dark-blue;
  }
}
