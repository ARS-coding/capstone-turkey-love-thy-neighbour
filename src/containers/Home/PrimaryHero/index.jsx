import React from "react";

import { Container, Row, Col } from "react-bootstrap";
import styled from "styled-components";
import CustomButton from "../../../components/CustomButton";

import "./index.scss";
import img from "../../../images/PrimaryHeroContentBgImg.png";

const CTAButton = styled(CustomButton)`
  color: var(--white);
  padding: 13px 65px;
  margin-top: 13px;
`;

function PrimaryHero() {
  return (
    <section className="snapping-section primary-hero-img-bg container-fluid">
      <Container className="primary-hero-content-container d-flex align-items-center">
        <Row className="primary-hero-content-row m-0">
          <Col
            xs={12}
            sm={12}
            md={6}
            className="first-column p-0 pb-xs-5 pb-md-5 d-flex flex-column justify-content-center align-items-start"
          >
            <h1>Love Thy Neighbor!</h1>
            <p>
              Letâ€™s build a strong, dynamic and caring community that uplifts
              each other.
            </p>
            <CTAButton className="cta-button">Join Us!</CTAButton>
            {/* TODO: Show sign up pop up when clicked */}
          </Col>
          <Col xs={12} sm={12} md={6} className="p-0 px-xs-1 px-md-1 h-100">
            <img src={img} alt="img" className="img-fluid" />
          </Col>
        </Row>
      </Container>
    </section>
  );
}

export default PrimaryHero;
