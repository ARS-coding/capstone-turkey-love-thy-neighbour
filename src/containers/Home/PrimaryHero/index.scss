@import "../../../style/variables";

.primary-hero-img-bg,
.primary-hero-content-container {
  height: calc(100vh - 60px);
}

.primary-hero-content-row,
.primary-hero-content-row > .col {
  width: 100%;
}

.primary-hero-img-bg {
  background-image: url("../../../utils/PrimaryHeroBgImg.png");

  h1 {
    color: $logo-blue;
    font-size: max(40px);
  }

  p {
    color: white;
    font-size: 20px;
  }

  img {
    width: 100%;
    height: 100%;
  }

  .first-column {
    @media screen and (max-width: 768px) {
      padding-bottom: 50px !important;
    }
  }

  .cta-button {
    background-color: $dark-blue;
  }
}
