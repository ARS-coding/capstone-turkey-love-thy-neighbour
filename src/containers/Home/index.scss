h1 {
  font-size: 36px;
}

// for sections and content containers
section, // all the sections inside home page
section > div // all the content containers inside home page
{
  height: calc(100vh - 60px);
  scroll-snap-align: start;
}

// section --> content container --> row
section > div > div.row {
  width: 100% !important;
}
