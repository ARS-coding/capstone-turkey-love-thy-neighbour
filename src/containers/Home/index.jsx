import React from "react";

import { Container } from "react-bootstrap";

import PrimaryHero from "../../components/PrimaryHero";
import SecondaryHero from "../../components/SecondaryHero";
import SocialProof from "../../components/SocialProof";
import Features from "../../components/Features";
import FAQ from "../../components/FAQ";
import Subscription from "../../components/Subscription";

import "./index.scss";

function Home() {
  return (
    <Container fluid className="section-wrapper">
      <PrimaryHero />
      <SecondaryHero />
      <SocialProof />
      <Features />
      <FAQ />
      <Subscription />
    </Container>
  );
}

export default Home;
