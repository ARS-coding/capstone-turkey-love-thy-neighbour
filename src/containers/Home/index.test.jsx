import React from "react";
import renderer from "react-test-renderer";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";
import Home from ".";

describe("Home page", () => {
    const mockState = {
        user: {
            firestoreDoc: {
                firstName: "Ali Riza",
                lastName: "Sahin",
                district: "KADIKÃ–Y",
                gender: "Male",
                age: 25,
                education: "High School",
                bio: "I am a freak.",
                interests: [{ content: "Fashion" }, { content: "Coding" }],
                number: "25",
                address: "my address",
                profileImageUrl: "",
                backgroundImageUrl: "",
            },
            authCred: {
                user: {
                    email: "Test Email",
                },
            },
        },
        popup: {
            isEditProfileOpen: false,
        }
    };
    const mockStore = configureStore();
    let store;
    let tree;

    beforeEach(() => {
        store = mockStore(mockState);
        tree = renderer
            .create(
                <Provider store={store}>
                    <Home />
                </Provider>
            )
            .toJSON();
    });

    it("renders Home correctly", () => {
        expect(tree).toMatchSnapshot();
    });
});
