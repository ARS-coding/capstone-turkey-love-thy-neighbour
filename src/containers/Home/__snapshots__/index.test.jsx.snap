// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page renders Home correctly 1`] = `
<div
  className="section-wrapper container-fluid"
>
  <section
    className="snapping-section primary-hero-img-bg container-fluid"
  >
    <div
      className="primary-hero-content-container d-flex align-items-center container"
    >
      <div
        className="primary-hero-content-row m-0 row"
      >
        <div
          className="first-column p-0 pb-2 pb-sm-2 pb-md-0 d-flex flex-column justify-content-center align-items-start col-md-6 col-sm-12 col-12"
        >
          <h1>
            Love Thy Neighbor!
          </h1>
          <p>
            Letâ€™s build a strong, dynamic and caring community that uplifts each other.
          </p>
          <button
            className="sc-bdnxRM sc-gtsrHT fhyKDU bLMOay"
            onClick={[Function]}
          >
            Join Us!
          </button>
        </div>
        <div
          className="px-0 ps-0 ps-sm-0 ps-md-3  col-md-6 col-sm-12 col-12"
        >
          <img
            alt="img"
            className="img-fluid"
            src="PrimaryHeroContentBgImg.png"
          />
        </div>
      </div>
    </div>
  </section>
  <section
    className="snapping-section secondary-hero-img-bg container-fluid"
  >
    <div
      className="secondary-hero-content-container d-flex align-items-center container"
    >
      <div
        className="secondary-hero-content-row m-0 row"
      >
        <div
          className="p-0 px-xs-1 px-md-1 h-100 d-flex justify-content-xs-start justify-content-sm-start justify-content-md-center align-items-md-center col-md-6 col-sm-12 col-12 order-md-first order-sm-last order-last"
        >
          <img
            alt="img"
            className="img-fluid"
            src="SecondaryHeroContentBgImg.png"
          />
        </div>
        <div
          className="px-0 pt-xs-5 pt-xs-5 pt-md-0 d-flex flex-column justify-content-center align-items-start col-md-6 col-sm-12 col-12 order-md-last order-sm-first order-first"
        >
          <h2>
            Welcome,
          </h2>
          <h1>
            to be welcomed.
          </h1>
          <p>
            Respect each other and feel the genuine 
            <span>
              connection.
            </span>
          </p>
        </div>
      </div>
    </div>
  </section>
  <section
    className="snapping-section social-proof-section container-fluid"
  >
    <div
      className="social-proof-content-container d-flex align-items-center container"
    >
      <div
        className="social-proof-content-row m-0 gap-2 row"
      >
        <div
          className="first-column col-md-5 col-sm-12 col-12"
        >
          <h2
            className="text-center text-sm-center text-md-start"
          >
            What our Neighbours say:
          </h2>
          <div
            className="accordion"
          >
            <div
              className="first-accordion-item accordion-item"
            >
              <h2
                className="d-flex align-items-center accordion-header"
              >
                <button
                  aria-expanded={true}
                  className="accordion-button"
                  onClick={[Function]}
                  type="button"
                >
                  <svg>
                    male-avatar.svg
                  </svg>
                  <div
                    className="d-flex flex-column"
                  >
                    <p
                      style={
                        Object {
                          "margin": "0",
                          "padding": "0",
                        }
                      }
                    >
                      Sammy Walson
                    </p>
                    <small
                      style={
                        Object {
                          "display": "block",
                          "margin": "0",
                          "padding": "0",
                        }
                      }
                    >
                      Istanbul, Uskudar
                    </small>
                  </div>
                </button>
              </h2>
              <div
                aria-expanded={null}
                className="accordion-collapse collapse show"
              >
                <div
                  className="accordion-body"
                >
                  It definitely has been a great experiance connecting with my neighbors on LTN.
                </div>
              </div>
            </div>
            <div
              className="second-accordion-item accordion-item"
            >
              <h2
                className="d-flex align-items-center accordion-header"
              >
                <button
                  aria-expanded={false}
                  className="accordion-button collapsed"
                  onClick={[Function]}
                  type="button"
                >
                  <svg>
                    female-avatar.svg
                  </svg>
                  <div
                    className="d-flex flex-column"
                  >
                    <p
                      style={
                        Object {
                          "margin": "0",
                          "padding": "0",
                        }
                      }
                    >
                      Violet Jenning
                    </p>
                    <small
                      style={
                        Object {
                          "display": "block",
                          "margin": "0",
                          "padding": "0",
                        }
                      }
                    >
                      Istanbul, Kadikoy
                    </small>
                  </div>
                </button>
              </h2>
              <div
                aria-expanded={null}
                className="accordion-collapse collapse"
              >
                <div
                  className="accordion-body"
                >
                  LTN, definitely is the right place to find like-minded people around you.
                </div>
              </div>
            </div>
            <div
              className="third-accordion-item accordion-item"
            >
              <h2
                className="d-flex align-items-center accordion-header"
              >
                <button
                  aria-expanded={false}
                  className="accordion-button collapsed"
                  onClick={[Function]}
                  type="button"
                >
                  <svg>
                    male-avatar.svg
                  </svg>
                  <div
                    className="d-flex flex-column"
                  >
                    <p
                      style={
                        Object {
                          "margin": "0",
                          "padding": "0",
                        }
                      }
                    >
                      Andrew Fitzgerald
                    </p>
                    <small
                      style={
                        Object {
                          "display": "block",
                          "margin": "0",
                          "padding": "0",
                        }
                      }
                    >
                      Istanbul, Avcilar
                    </small>
                  </div>
                </button>
              </h2>
              <div
                aria-expanded={null}
                className="accordion-collapse collapse"
              >
                <div
                  className="accordion-body"
                >
                  Had a lot of chances to connect with my community... Amazing app!
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="second-column mt-xs-4 mt-sm-4 mt-md-0 col-md-5 col-sm-12 col-12"
        >
          <h2
            className="text-center text-sm-center text-md-start"
          >
            When our Neighbours met:
          </h2>
          <div
            className="carousel slide"
            onKeyDown={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
          >
            <div
              className="carousel-indicators"
            >
              <button
                aria-current={true}
                aria-label="Slide 1"
                className="active"
                data-bs-target=""
                onClick={[Function]}
                type="button"
              />
              <button
                aria-current={false}
                aria-label="Slide 2"
                data-bs-target=""
                onClick={[Function]}
                type="button"
              />
              <button
                aria-current={false}
                aria-label="Slide 3"
                data-bs-target=""
                onClick={[Function]}
                type="button"
              />
            </div>
            <div
              className="carousel-inner"
            >
              <div
                className="active carousel-item"
              >
                <img
                  alt="First meeting slide"
                  className="d-block w-100 h-100"
                  src="meeting-one.png"
                />
              </div>
              <div
                className="carousel-item"
              >
                <img
                  alt="Second meeting slide"
                  className="d-block w-100 h-100"
                  src="meeting-two.png"
                />
              </div>
              <div
                className="carousel-item"
              >
                <img
                  alt="Third meeting slide"
                  className="d-block w-100 h-100"
                  src="meeting-three.png"
                />
              </div>
            </div>
            <a
              className="carousel-control-prev"
              href="#"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
            >
              <span
                aria-hidden="true"
                className="carousel-control-prev-icon"
              />
              <span
                className="visually-hidden"
              >
                Previous
              </span>
            </a>
            <a
              className="carousel-control-next"
              href="#"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
            >
              <span
                aria-hidden="true"
                className="carousel-control-next-icon"
              />
              <span
                className="visually-hidden"
              >
                Next
              </span>
            </a>
          </div>
        </div>
      </div>
    </div>
  </section>
  <section
    className="features-section container-fluid"
  >
    <div
      className="features-content-container d-flex flex-column justify-content-center container"
    >
      <h2>
        Features
      </h2>
      <div
        className="features-content-row d-md-flex justify-content-md-around text-center row"
      >
        <div
          className="col-md-3 col-sm-12 col-12"
        >
          <div
            className="d-flex flex-column align-items-center justify-content-center card"
          >
            <svg
              className="network-svg img-fluid"
            >
              network.svg
            </svg>
            <div
              className="card-title h5"
            >
              Meet
            </div>
            <p
              className="card-text"
            >
              Find people in your neighborhood and expand your network.
            </p>
          </div>
        </div>
        <div
          className="my-1 my-sm-2 my-md-0 col-md-3 col-sm-12 col-12"
        >
          <div
            className="d-flex flex-column align-items-center justify-content-center card"
          >
            <svg
              className="speech-bubble-svg img-fluid"
            >
              speech-bubble.svg
            </svg>
            <div
              className="card-title h5"
            >
              Filtering
            </div>
            <p
              className="card-text"
            >
              Filter nearby people by their age, gender and interests.
            </p>
          </div>
        </div>
        <div
          className="col-md-3 col-sm-12 col-12"
        >
          <div
            className="d-flex flex-column align-items-center justify-content-center card"
          >
            <svg
              className="coupon-svg img-fluid"
            >
              coupon.svg
            </svg>
            <div
              className="card-title h5"
            >
              Coupons
            </div>
            <p
              className="card-text"
            >
              Get coupons from our sponsored places and pay less.
            </p>
          </div>
        </div>
      </div>
    </div>
  </section>
  <section
    className="container-fluid"
  >
    <div
      className="faq-content-container d-flex align-items-center container"
    >
      <div
        className="faq-content-row m-0 row"
      >
        <div
          className="first-column d-flex flex-column justify-content-center align-items-start col-md-6 col-sm-12 col-12"
        >
          <h2>
            FAQs
          </h2>
          <h5>
            Are meetings safe?
          </h5>
          <p>
            For this important topic, we have collaborated with popular cafes and places all around of Istanbul so that our neighbours would be able to meet with each other without worries.
          </p>
          <h5>
            What is the point?
          </h5>
          <p>
            More than 264 million people suffer from depression worldwide and with quarantine, it was not getting any better. Thatâ€™s why itâ€™s so important for us to solve the problem of people feeling more and more isolated as time progresses. We care about you, and would like you to care about your neighbour so that, we could be a stronger community.
          </p>
          <h5>
            How does all of this work?
          </h5>
          <p>
            With the data that you provide in your profile, we are carefully filtering neighbours that you may want to meet and network with.
          </p>
        </div>
        <div
          className="my-md-auto d-flex d-sm-flex d-md-block justify-content-center justify-content-sm-center justify-content-md-center col-md-6 col-sm-12 col-12"
        >
          <img
            alt="img"
            className="img-fluid"
            src="UndrawQuestion.png"
          />
        </div>
      </div>
    </div>
  </section>
  <section
    className="subscription-img-bg container-fluid"
  >
    <div
      className="subscription-content-container d-flex align-items-center container"
    >
      <div
        className="subscription-content-row m-0 row"
      >
        <div
          className="first-column d-flex flex-column justify-content-center align-items-start col-md-6 col-sm-12 col-12"
        >
          <h1>
            Subscribe to our 
            <br />
             newsletter.
          </h1>
          <h3>
            Never miss an update !
          </h3>
          <p>
            Signup to our newsletter to get a weekly digest.
          </p>
          <div
            className="button-input-wrapper"
          >
            <input
              aria-describedby="emailHelp"
              className="form-email"
              onChange={[Function]}
              placeholder="E-Mail"
              type="email"
              value=""
            />
            <button
              className="btn btn-primary"
              disabled={true}
              type="submit"
            >
              <svg>
                Check.svg
              </svg>
            </button>
          </div>
        </div>
        <div
          className="my-md-auto d-flex d-sm-flex d-md-block justify-content-center justify-content-sm-center justify-content-md-center col-md-6 col-sm-12 col-12"
        >
          <img
            alt="phoneimg"
            className="img-fluid"
            src="PhoneandNotifications.svg"
          />
        </div>
      </div>
    </div>
  </section>
</div>
`;
